test/responders/gif_me_test.exs

```diff
@@ -0,0 +1,13 @@
+defmodule Elix.Responders.GifMeTest do
+  use Hedwig.RobotCase
+  use ExVCR.Mock, adapter: ExVCR.Adapter.Hackney
+
+  @tag start_robot: true, name: "elix", responders: [{Elix.Responders.GifMe, []}]
+  test "gif me - responds with a GIF url", %{adapter: adapter, msg: msg} do
+    use_cassette "giphy_search_steve" do
+      send adapter, {:message, %{msg | text: "gif me steve"}}
+      assert_receive {:message, %{text: text}}
+      assert String.ends_with?(text, ".gif")
+    end
+  end
+end
```

Comment:
Very pro the `GiphyClientMock`. Having the robot be able to tell you when something goes wrong with your integration, either via external instrumentation or with just the error message as a response seems like a reasonable way to find out your integration is broken.

> if we extract the HTTP request-making logic into its own module, we'd _still_ need something like `ex_vcr` to test that, right?

Why is that? It seems like you have a known good case or cases, and you program for them, and in the case you meet a case you don't know about, crash and report?